### PR TITLE
report_* models to toucan2

### PR DESCRIPTION
**Pull request by qnkhuat**
Convert Dashboard, DashboardCard, DashboardCardSeries to toucan 2, the new model will be keywords with a rprefix `:model/` (Dashboard => :model/Dashboard)

Since the model symbol is used in many places and replacing all `Dashboard` => `:model/Dashboard` will be a huge change across the code base, so my general strategy is to create a temp symbol `(def Dashboard :model/Dashboard)`, so that most places that use the `Dashboard` symbol can stay the same.

But to verify my changes work, I'll change `Dashboard` => `:model/Dashboard` in its test namespaces: API and model testing.

Note for reviewers: it’s easier to review by commit

Epic: https://github.com/metabase/metabase/issues/29231

This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/30068)

### CODE REVIEW

File: test/metabase/api/dashboard_test.clj
Comment (qnkhuat) [inline test/metabase/api/dashboard_test.clj:1]: mostly replacing the old symbol model to its new equivalent keywords and fix the indentation

File: test/metabase/models/dashboard_card_test.clj
Comment (qnkhuat) [inline test/metabase/models/dashboard_card_test.clj:1]: mostly replacing the old symbol model to its new equivalent keywords and fix the indentation

File: src/metabase/models/dashboard.clj [hunk 1]
```diff
@@ -30,69 +30,49 @@
    [metabase.util.i18n :as i18n :refer [tru]]
    [metabase.util.log :as log]
    [metabase.util.schema :as su]
+   [methodical.core :as methodical]
    [schema.core :as s]
    [toucan.hydrate :refer [hydrate]]
-   [toucan.models :as models]
    [toucan2.core :as t2]))
 
-;;; --------------------------------------------------- Hydration ----------------------------------------------------
```
Comment (qnkhuat) [inline src/metabase/models/dashboard.clj:38]: these hydration methods are unchanged, just moved down

File: src/metabase/models/dashboard.clj [hunk 2]
```diff
@@ -30,69 +30,49 @@
...
-                                                 :left-join [[:collection :collection] [:= :collection.id :dashboard.collection_id]]
-                                                 :where     [:in :dashboard.id (into #{} (map u/the-id) dashboards)]}))]
-      (for [dashboard dashboards]
-        (assoc dashboard :collection_authority_level (get coll-id->level (u/the-id dashboard)))))))
+(def Dashboard
+  "Used to be the toucan1 model name defined using [[toucan.models/defmodel]], not it's a reference to the toucan2 model name.
+   We'll keep this till we replace all the Dashboard symbol in our codebase."
+  :model/Dashboard)
 
-(comment moderation/keep-me)
```
Comment (qnkhuat) [inline src/metabase/models/dashboard.clj:70]: it's not removed, just moved down.

File: src/metabase/models/dashboard_card.clj [hunk 3]
```diff
@@ -42,19 +59,6 @@
     (apply set/union (mi/perms-objects-set card read-or-write) (for [series-card series]
                                                                  (mi/perms-objects-set series-card read-or-write)))))
 
-(defn- pre-insert [dashcard]
-  (let [defaults {:parameter_mappings     []
-                  :visualization_settings {}}]
-    (merge defaults dashcard)))
```
Comment (metamben) [inline src/metabase/models/dashboard_card.clj:48]: Why is this thrown away?

File: src/metabase/models/dashboard_card_series.clj [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
   (:require
    [metabase.models.serialization :as serdes]
-   [toucan.models :as models]
+   [methodical.core :as methodical]
    [toucan2.core :as t2]))
 
-(models/defmodel DashboardCardSeries :dashboardcard_series)
+(def DashboardCardSeries
+  "Used to be the toucan1 model name defined using [[toucan.models/defmodel]], not it's a reference to the toucan2 model name.
+   We'll keep this till we replace all the DashboardCard symbol in our codebase."
```
Comment (metamben) [inline src/metabase/models/dashboard_card_series.clj:9]: `DashboardCard` -> `DashboardCardSeries`

File: src/metabase/models/dashboard_card.clj (hunk 3 above)
Comment (qnkhuat) [reply src/metabase/models/dashboard_card.clj:48]: it's not, you can see the `define-before-insert` is doing the same thing.

File: src/metabase/models/dashboard_card_series.clj (hunk 4 above)
Comment (qnkhuat) [inline src/metabase/models/dashboard_card_series.clj:9]:
```suggestion
   We'll keep this till we replace all the DashboardCardSeries symbol in our codebase."
```

**Review by metamben — COMMENTED** [review]
> Reviewed 2 of 5 files at r3, 10 of 10 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30068)* status: all files reviewed, 7 unresolved discussions (waiting on @qnkhuat)

___
*[`src/metabase/models/dashboard_card.clj` line 47 at r5](https://reviewable.io/reviews/metabase/metabase/30068#-NUSLiBz5XyC4ApCNFre:-NUSLiC-2U4FD0MMowxr:b-6sh1zz) ([raw file](https://github.com/metabase/metabase/blob/358e8b8df39e8087411e6a90830c5061a6aebe3a/src/metabase/models/dashboard_card.clj#L47)):*
> ```Clojure
>  [dashcard]
>  (merge {:parameter_mappings     []
>          :visualization_settings {}} dashcard))
> ```

A line break before `dashcard` would be the usual formatting, I think.

**Review by metamben — APPROVED** [review]
LGTM

**Review by qnkhuat — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/metabase/metabase/30068)* status: 10 of 13 files reviewed, 6 unresolved discussions (waiting on @metamben and @qnkhuat)

*[a discussion](https://reviewable.io/reviews/metabase/metabase/30068#gh-1173047205:-NV89OAx8owCzmmAcuqL:b-896fix) (no related file):*
<details><summary><i>Previously, metamben wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

`DashboardCard` -> `DashboardCardSeries`
</blockquote></details>

Done.

___

**Pull request merged**